examples/helloworld/src/index.js

```diff
@@ -1,4 +1,26 @@
 import React from 'react';
+import _ from 'underscore';
 import App from './App';
 
-React.render(<App />, document.getElementById('root'))
+
+var state = {
+    very: {
+        deeply: {
+            nested: {
+                counts: _.range(400).map(function () { return 0; })
+            }
+        }
+    }
+};
+
+function swapper(f) {
+    state = f(state);
+    queueRender();
+}
+
+function queueRender() {
+    var cur = ReactCursor.Cursor.build(state, swapper);
+    React.render(<App cursor={cur} />, document.getElementById('root'));
```

Comment:
In your example, the root cursor instance is changed on each render. It leads the outer reference to the root cursor object be lost. However, it seems it can be access the root cursor via `cmp.props.cursor` because the root cursor is a props of component now.